### PR TITLE
GSOC 2024 proposal

### PR DESCRIPTION
**Pull request by rareddy**
Add project ideas for Kubeflow's application to Google Summer of Code 2024, aiming to participate as an organization and recruit student contributors

### CODE REVIEW

**Comment by rareddy** [comment]
@jbottum @thesuperzapper can you approve this please?

File: content/en/events/gsoc-2024.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+**Difficulty**: medium
+
+**Length**: [250 hrs]
+
+**Mentors**: Andrey Velichkevich, Johnu George, Yuki Iwai
+
+---
+---
+
+### Project 6: ush-based metrics collection for Katib
```
Comment (jbottum) [inline content/en/events/gsoc-2024.md:136]: should this be Push-based?

File: content/en/events/gsoc-2024.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+## Students
+
+Welcome to an incredible chance to dive into the vibrant Kubeflow community! Join us and become a valued contributor to our rapidly expanding network dedicated to AI/ML technologies on Kubernetes. Kubeflow mentors come with years of valuable experience and a passion for community growth and project enhancement, they're the guiding stars of our program. Immerse yourself in a dynamic environment where you'll not only gain insights from top industry experts but also sharpen your skills in open source collaboration, time management, and presentation prowess alongside making impactful technical contributions to projects of your interest. Our dedicated mentors will be by your side every step of the way, offering invaluable guidance and support throughout the project. Get ready to embark on an exciting journey of growth and learning!
+
+**If you need to reach out our mentors, join [Kubeflow Slack](https://kubeflow.slack.com/archives/C7REE0ETX) and look up your mentor or find project specific channel on Kubeflow slack**
```
Comment (thesuperzapper) [inline content/en/events/gsoc-2024.md:25]: Probably use the Kubeflow slack website link (makes it easier for people to get invited by the bot).

https://www.kubeflow.org/docs/about/community/#kubeflow-slack

File: content/en/events/gsoc-2024.md (hunk 1 above)
Comment (rareddy) [reply content/en/events/gsoc-2024.md]: good catch, corrected it now.

File: content/en/events/gsoc-2024.md (hunk 2 above)
Comment (rareddy) [reply content/en/events/gsoc-2024.md]: agree, updated the link

**Review by Arhell — COMMENTED** [review]
/ok-to-test

**Comment by rareddy** [comment]
Can we get this PR approved and merged please this is sensitive as I need to provide a URL to link to GSoC program.

File: content/en/events/gsoc-2024.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+**Mentors**: Andrey Velichkevich, Johnu George, Yuki Iwai
+
+---
+---
+
+### Project 6: Push-based metrics collection for Katib
+
+Open issue: [https://github.com/kubeflow/katib/issues/577](https://github.com/kubeflow/katib/issues/577).
+
+Katib implements Metrics Collector as a sidecar container to collect training metrics from the Trials once training is complete. This Metrics Collector waits until the training container is complete and parses training logs to get appropriate metrics like accuracy or loss to get evaluation results for the HyperParemeter tuning algorithm.
```
Comment (andreyvelich) [inline content/en/events/gsoc-2024.md:143]:
```suggestion
Katib implements Metrics Collector as a sidecar container to collect training metrics from the Trials once training is complete. This Metrics Collector waits until the training container is complete and parses training logs to get appropriate metrics like accuracy or loss to get evaluation results for the HyperParameter tuning algorithm.
```

**Comment by thesuperzapper** [comment]
@rareddy can you please remove people's emails from the PR, as we don't want to publish those?

Also while you are doing it, fix that small typo raised by @andreyvelich, otherwise its find to merge for now, and we can update later.

After this, we can put a more prominent link in the "docs" section to the "events" section, probably in the top-level of the sidebar, rather than only under the "about" section.

File: content/en/events/gsoc-2024.md (hunk 3 above)
Comment (rareddy) [reply content/en/events/gsoc-2024.md]: fixed now.

**Comment by rareddy** [comment]
> rather than only under the "about" section (or at least adding an "events" link to the top-bar)

@thesuperzapper so move the "events" to the same level "about" in side bar.  Should I replace "kubeflow summit" one with  "events" too perhaps?

**Comment by thesuperzapper** [comment]
@rareddy we can probably just comment out the Kubeflow Summit one for now, and replace it with the Kubecon EU 2024 (once we have a page for that).

But we can add an "Events" button up there, and possibly rename "Documentation" to "Docs" for less space usage (but lets do that in a separate PR).

File: content/en/events/gsoc-2024.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+**Length:** 250 hrs
+
+**Mentors:** Mathew Wicks, Kimonas Sitorchos, Julius von Kohout
+
+---
+---
+
+###  Project 4: Implement LLM Tuning API for Katib
+
+Recently, we implemented [https://github.com/kubeflow/training-operator/blob/master/docs/proposals/train_api_proposal.md](a new `train` Python SDK API) in Kubeflow Training Operator to easily fine-tune LLMs on multiple GPUs with predefined datasets provider, model provider, and HuggingFace trainer.
```
Comment (thesuperzapper) [inline content/en/events/gsoc-2024.md:106]: @rareddy This link is around the wrong way, it should be `[a new `train` Python SDK API](https://github.com/kubeflow/training-operator/blob/master/docs/proposals/train_api_proposal.md)`

**Pull request merged**